Running v1 and beta in parallel causes master branch to be out of sync

See the error below:

```
Starting: push clean beta metadata to msgraph-metadata repo
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.180.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
========================== Starting Command Output ===========================
/usr/bin/pwsh -NoLogo -NoProfile -NonInteractive -Command . '/home/vsts/work/_temp/668c71c9-e0b1-4ceb-978b-8a45be851d03.ps1'
About to add clean beta metadata file.....
[master d1df8a2] Update clean metadata file with 39906 [skip ci]
 2 files changed, 661 insertions(+), 283 deletions(-)
Added and commited cleaned beta metadata.
To https://github.com/microsoftgraph/msgraph-metadata
 ! [rejected]        master -> master (fetch first)
error: failed to push some refs to 'https://github.com/microsoftgraph/msgraph-metadata'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
Pushed the results of the build 39906 to the master branch.
##[error]PowerShell exited with code '1'.
Finishing: push clean beta metadata to msgraph-metadata repo
```

https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=39906&view=logs&j=4c81e8e4-6eb9-5def-7261-b44ad0fc1b7d&t=163839c6-f69e-5b5d-39e2-5b38e97d3fe3&l=25

We should have a `git pull` before `git push`. 
[AB#8072](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/8072)

--- a/scripts/git-push-cleanmetadata.ps1
+++ b/scripts/git-push-cleanmetadata.ps1
@@ -6,6 +6,9 @@ if (!$env:PublishChanges)
 
 Write-Host "About to add clean $env:EndpointVersion metadata file....."
 
+# checkout master to move from detached HEAD mode
+git checkout master
+
 git add . | Write-Host
 if ($env:BUILD_REASON -eq 'Manual') # Skip CI if manually running this pipeline.
 {
@@ -17,11 +20,6 @@ else
 }
 
 Write-Host "Added and commited cleaned $env:EndpointVersion metadata." -ForegroundColor Green
-
-# sync branch before pushing
-# this is especially important while running v1 and beta in parallel
-# and one process goes out of sync because of the other's check-in
-git pull origin master
 
 git push --set-upstream origin master | Write-Host
 Write-Host "Pushed the results of the build $env:BUILD_BUILDID to the master branch." -ForegroundColor Green

--- a/scripts/git-push-cleanmetadata.ps1
+++ b/scripts/git-push-cleanmetadata.ps1
@@ -21,5 +21,10 @@ else
 
 Write-Host "Added and commited cleaned $env:EndpointVersion metadata." -ForegroundColor Green
 
+# sync branch before pushing
+# this is especially important while running v1 and beta in parallel
+# and one process goes out of sync because of the other's check-in
+git pull origin master
+
 git push --set-upstream origin master | Write-Host
 Write-Host "Pushed the results of the build $env:BUILD_BUILDID to the master branch." -ForegroundColor Green

--- a/scripts/git-push-cleanmetadata.ps1
+++ b/scripts/git-push-cleanmetadata.ps1
@@ -18,5 +18,10 @@ else
 
 Write-Host "Added and commited cleaned $env:EndpointVersion metadata." -ForegroundColor Green
 
+# sync branch before pushing
+# this is especially important while running v1 and beta in parallel
+# and one process goes out of sync because of the other's check-in
+git pull origin master
+
 git push --set-upstream origin master | Write-Host
 Write-Host "Pushed the results of the build $env:BUILD_BUILDID to the master branch." -ForegroundColor Green

